### ISSUE TITLE
Leading `+` character when parsing `amount`

### ISSUE THREAD

**Issue by antonok-edm** [solana-labs/solana-pay#101]
@chitalian and I are working on [a Rust port](https://gitlab.com/solsticepay/solana-pay) of the `@solana/pay` JavaScript SDK. Many float implementations, including the one in the Rust standard library and the `BigNumber` library used by the official SDK, consider a leading `+` to be valid when parsing numbers. We [noticed](https://gitlab.com/solsticepay/solana-pay/-/issues/4#note_871188197) that the [description in the spec](https://github.com/solana-labs/solana-pay/blob/master/SPEC.md#amount) does not address the behavior of a leading `+` character. For reference, it looks like the official SDK happens to forbid a leading `+`, although it's unclear from [the code](https://github.com/solana-labs/solana-pay/blob/master/core/src/parseURL.ts#L56) whether or not it's intentional. Forbidding a leading `+` feels like the correct choice, but it would be great to have that clarified in the spec.

**Comment by jordansexton** [comment]
Yes, forbidding a leading plus would be good. This was intentional in the reference implementation, but we could stand to make the spec more strict.

Also, awesome project! I can't view your Gitlab, so I'm guessing it's still private?

**Comment by jordansexton** [comment]
Since you're doing an awesome job of reviewing the spec and finding places where it's ambiguous or inconsistent, would you and @chitalian take a look at #77 and the [latest spec](https://github.com/solana-labs/solana-pay/blob/link-request/SPEC.md) there?

Transfer requests should be mostly the same, but transaction requests are new. Wallets are actively implementing this now, but it's still open for feedback!

**Comment by antonok-edm** [comment]
> Yes, forbidding a leading plus would be good. This was intentional in the reference implementation, but we could stand to make the spec more strict.

Sounds good :smile: 

> Also, awesome project! I can't view your Gitlab, so I'm guessing it's still private?

Yep, we're in the never-ending "almost ready to publish" phase :sweat_smile:

We'll be submitting it as an entry in the Riptide hackathon though, so it'll be up soon! You're welcome to add it to the `Coming soon` section `Supporting Wallets` in the meantime - we're using it for transfer requests.

> would you and @chitalian take a look at https://github.com/solana-labs/solana-pay/pull/77 and the [latest spec](https://github.com/solana-labs/solana-pay/blob/link-request/SPEC.md) there?

I took a quick scan over it, I will add my thoughts there.
